pallets/pallet-governance/src/tests.rs

```diff
@@ -0,0 +1,437 @@
+			assert_eq!(CurrentAuthorities::<Runtime>::get(), (11..21).collect::<BTreeSet<_>>());
+
+			assert_eq!(
+				Votes::<Runtime>::get(first_proposal),
+				CastedVotes { yays: Default::default(), nays: Default::default() }
+			);
+			assert_eq!(ProposalsToResolve::<Runtime>::get(), Default::default());
+			assert_eq!(Proposals::<Runtime>::get(first_proposal), None);
+			assert_eq!(
+				Expiry::<Runtime>::get(System::block_number() + EXPIRY_PERIOD),
+				Default::default()
+			);
+		});
+}
+
+#[test]
+fn can_reject_council_rotate_authorities() {
+	MockGenesisConfig::with_authorities((11..21).collect::<Vec<_>>())
+		.build()
+		.execute_with(|| {
+			// Setup data
+			let call = Box::new(RuntimeCall::Governance(crate::Call::council_rotate_authorities {
+				new_members: (21..31).collect::<Vec<_>>(),
+			}));
+			let authority_member: AccountId = 11;
+			let first_proposal = 1u32;
+
+			// Propose
+			assert_ok!(Governance::initiate_proposal(
+				RuntimeOrigin::signed(authority_member),
+				call.clone()
+			));
+
+			// Vote 2 tickets rejected.
+			for i in 12..14 {
+				assert_ok!(Governance::vote(RuntimeOrigin::signed(i), first_proposal, false));
+			}
+
+			// dispatch
+			Governance::on_finalize(System::block_number());
+
+			// Verify the authorities are set.
+			assert_eq!(CurrentAuthorities::<Runtime>::get(), (11..21).collect::<BTreeSet<_>>());
+
+			assert_eq!(
+				Votes::<Runtime>::get(first_proposal),
+				CastedVotes { yays: Default::default(), nays: Default::default() }
+			);
+			assert_eq!(ProposalsToResolve::<Runtime>::get(), Default::default());
+			assert_eq!(Proposals::<Runtime>::get(first_proposal), None);
+
+			System::assert_has_event(RuntimeEvent::Governance(
+				Event::<Runtime>::ProposalRejected {
+					id: first_proposal,
+					reason: RejectReason::ByVoting,
+				},
+			));
+		});
+}
+
+#[test]
+fn can_resolve_after_proposal() {
+	MockGenesisConfig::with_authorities(vec![11]).build().execute_with(|| {
+		let call = Box::new(RuntimeCall::Governance(crate::Call::council_rotate_authorities {
+			new_members: (21..31).collect::<Vec<_>>(),
+		}));
+		let authority_member: AccountId = 11;
+		let first_proposal = 1u32;
+
+		// Test the function initiate_proposal works.
+		assert_ok!(Governance::initiate_proposal(
+			RuntimeOrigin::signed(authority_member),
+			call.clone()
+		));
+
+		// Dispatch
+		Governance::on_finalize(System::block_number());
+
+		// Verify the authorities are set.
+		assert_eq!(CurrentAuthorities::<Runtime>::get(), (21..31).collect::<BTreeSet<_>>());
+
+		assert_eq!(Proposals::<Runtime>::get(first_proposal), None);
+
+		assert_eq!(
+			Votes::<Runtime>::get(first_proposal),
+			CastedVotes { yays: Default::default(), nays: Default::default() }
+		);
+
+		let last_event = System::events().last().unwrap().event.clone();
+		matches!(last_event, RuntimeEvent::Governance(Event::<Runtime>::ProposalPassed {
+			id,
+			call,
+			result: Ok(..),
+		}) if id == first_proposal && call == call );
+	});
+}
```

Comment:
add a test to ensure cannot vote on a proposal after it is passed/rejected/expired